squad/api/rest.py

```diff
@@ -401,6 +404,39 @@ def test_results(self, request, pk=None):
         )
         return Response(serializer.data)
 
+    @detail_route(methods=['post'], suffix='subscribe')
+    def subscribe(self, request, pk=None):
+        subscriber_email = request.data.get("email", None)
+        if subscriber_email is None:
+            raise serializers.ValidationError("'email' field is mandatory.")
+        try:
+            validate_email(subscriber_email)
+        except core_exceptions.ValidationError:
+            raise serializers.ValidationError("Invalid 'email' field value.")
+        user = User.objects.filter(email=subscriber_email).first()
+        if user is None:
+            sub, _ = Subscription.objects.get_or_create(email=subscriber_email, project=self.get_object())
+        else:
+            sub, _ = Subscription.objects.get_or_create(user=user, project=self.get_object())
+        data = {"email": subscriber_email}
+        return Response(data, status=status.HTTP_201_CREATED)
+
+    @detail_route(methods=['post'], suffix='unsubscribe')
+    def unsubscribe(self, request, pk=None):
+        subscriber_email = request.data.get("email", None)
+        if subscriber_email is None:
+            raise serializers.ValidationError("'email' field is mandatory.")
+        try:
+            validate_email(subscriber_email)
+        except core_exceptions.ValidationError:
+            raise serializers.ValidationError("Invalid 'email' field value.")
+        subs = Subscription.objects.filter(Q(email=subscriber_email) | Q(user__email=subscriber_email))
+        if subs.exists():
+            for sub in subs:
+                sub.delete()
```

Comment:
you don't need to check for emptiness, or to loop over the items. just call `subs.delete()` and it will do the right thing.

Reply:
oh, I didn't know :)